octodns/provider/base.py

```diff
@@ -30,6 +32,24 @@ def __init__(self, id, apply_disabled=False,
         self.delete_pcent_threshold = delete_pcent_threshold
         self.strict_supports = strict_supports
 
+    def _check_dynamic_for_private_ips(self, desired, record):
+        if record._type not in ('A', 'AAAA'):
+            # won't necessarily contain IPs
+            return
+        for name, pool in record.dynamic.pools.items():
+            for i, value in enumerate(pool.data['values']):
+                ip = ip_address(value['value'])
+                status = value['status']
+                if not ip.is_global and status != 'up':
```

Comment:
I think we should allow private IPs with `status: down` instead of forcing them `up`. Health-check is only present if `status: obey` so only `obey`s should be forced `up`.